core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreferWritePartitioning.java

```diff
@@ -46,6 +49,28 @@ public Result apply(TableWriterNode node, Captures captures, Context context)
             return Result.empty();
         }
 
+        int minimumNumberOfPartitions = getPreferredWritePartitioningMinNumberOfPartitions(context.getSession());
```

Comment:
add a comment that this allows to use preferred write partitioning even if stats are missing or broken